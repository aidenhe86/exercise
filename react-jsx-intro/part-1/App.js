const App = () => (
  <div>
    <FirstComponent />
    <NamedComponent name="Aiden" />
  </div>
);
