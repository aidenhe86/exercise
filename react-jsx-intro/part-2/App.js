const App = () => (
  <div>
    <Tweet
      name="Example1"
      username="ex1"
      message="THIS IS EX1!"
      date={new Date().toString()}
    />
    <br></br>
    <Tweet
      name="Example2"
      username="ex2"
      message="THIS IS EX2!"
      date={new Date().toString()}
    />
    <br></br>
    <Tweet
      name="Example3"
      username="ex3"
      message="THIS IS EX3!"
      date={new Date().toString()}
    />
  </div>
);
