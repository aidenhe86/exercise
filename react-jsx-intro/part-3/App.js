const App = () => (
  <div>
    <Person name="ex1" age={18} hobbies={["read", "write"]} />
    <Person name="ex2" age={16} hobbies={["read", "text to people"]} />
    <Person name="ex3" age={56} hobbies={["read", "listen music"]} />
  </div>
);
