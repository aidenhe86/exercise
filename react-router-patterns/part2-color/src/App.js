import React from "react";
import ColorRoute from "./ColorRoute";
import { BrowserRouter } from "react-router-dom";
import "./App.css";

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <ColorRoute />
      </BrowserRouter>
    </div>
  );
}

export default App;
