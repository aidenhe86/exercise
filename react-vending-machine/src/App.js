import React from "react";
import VendingMachine from "./VendingMachine";
import Soda from "./Soda";
import Candy from "./Candy";
import Chips from "./Chips";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import "./App.css";

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<VendingMachine />}></Route>
          <Route path="soda" element={<Soda />} />
          <Route path="candy" element={<Candy />} />
          <Route path="chips" element={<Chips />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
