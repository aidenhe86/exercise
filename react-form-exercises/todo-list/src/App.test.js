import { render, screen } from "@testing-library/react";
import App from "./App";

// default smoke test
test("renders without crashing", () => {
  render(<App />);
});

// Snapshot test
test("matches snapshot", function () {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
