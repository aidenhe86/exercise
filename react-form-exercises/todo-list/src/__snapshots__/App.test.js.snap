// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="TodoList"
    >
      <h1>
        Todo List
      </h1>
      <form>
        <label
          for="task"
        >
          Task: 
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
        <button>
          Add a new Todo!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
